webapp/src/js/services/auth.js

```diff
@@ -135,11 +135,9 @@ angular.module('inboxServices').factory('Auth',
             .map(function(permissions, key) {
               return checkPermissions(requiredPermissions[key], disallowedPermissions[key], roles, settings);
             })
-            .filter(function(result) {
-              return result === true;
-            });
+            .includes(true);
```

Comment:
If I understand this we can use `array.some` instead of `map` and `includes`. eg:

```js
const allowed = permissionsList.some((permission, index) => {
  return checkPermissions(requiredPermissions[index], disallowedPermissions[index], roles, settings);
});
```

That way the code knows it can stop checking as soon as one of them returns `true`, and you don't have to create a whole new array of booleans before throwing it away again.

Reply:
Something interesting I have tried about `Array.prototype.some()`. It will behave correctly if the results being evaluated explicitly return only `boolean` i.e. true or false . The functionality being tested above will return `'missing required permission(s)'` or `'found disallowed permission(s)'` if validation fails. Thus `Array.prototype.some()` will always result to True

Reply:
Ok, makes sense. I guess add change the `some` function to...

```js
const allowed = permissionsList.some((permission, index) => {
  return true === checkPermissions(requiredPermissions[index], disallowedPermissions[index], roles, settings);
});
```